[reland] minor improvement to dist quantization

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #67401
* #67400

reland of https://github.com/pytorch/pytorch/pull/66649
some minor changes to dist quantization, mainly change the namespace and add some notes for future code dedup

Differential Revision: [D31979269](https://our.internmc.facebook.com/intern/diff/D31979269/)

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

Commit:
[BE] minor improvement to dist quantization

some minor changes to dist quantization, mainly change the namespace and add some notes for future code dedup
ghstack-source-id: 141336191

Differential Revision: [D31979269](https://our.internmc.facebook.com/intern/diff/D31979269/)

[ghstack-poisoned]

mrshenli reviewed (commented):
Stamp to unblock. Please wait for the tests added by "ci/master" tag.

Commit:
Update on "[reland] minor improvement to dist quantization"


reland of https://github.com/pytorch/pytorch/pull/66649
some minor changes to dist quantization, mainly change the namespace and add some notes for future code dedup

Differential Revision: [D31979269](https://our.internmc.facebook.com/intern/diff/D31979269/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

Commit:
Update on "[reland] minor improvement to dist quantization"


reland of https://github.com/pytorch/pytorch/pull/66649
some minor changes to dist quantization, mainly change the namespace and add some notes for future code dedup

Differential Revision: [D31979269](https://our.internmc.facebook.com/intern/diff/D31979269/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

Commit:
Update on "[reland] minor improvement to dist quantization"


reland of https://github.com/pytorch/pytorch/pull/66649
some minor changes to dist quantization, mainly change the namespace and add some notes for future code dedup

Differential Revision: [D31979269](https://our.internmc.facebook.com/intern/diff/D31979269/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

Commit:
Update on "[reland] minor improvement to dist quantization"


reland of https://github.com/pytorch/pytorch/pull/66649
some minor changes to dist quantization, mainly change the namespace and add some notes for future code dedup

Differential Revision: [D31979269](https://our.internmc.facebook.com/intern/diff/D31979269/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]
